src/extra/AutoUi/Collection/index.tsx

```diff
@@ -231,7 +233,13 @@ export const AutoUICollection = <T extends AutoUIBaseResource<T>>({
 	);
 
 	React.useEffect(() => {
-		setSelected((oldSelected) => getSelectedItems(oldSelected, filtered));
+		setFilteredSelected(getSelectedItems(selected, filtered));
+	}, [selected, filtered]);
+
+	React.useEffect(() => {
+		setFilteredSelected((oldFilteredSelected) =>
+			getSelectedItems(oldFilteredSelected, filtered),
+		);
```

Comment:
Since we don't want to change the selection while filtering, the entire useEffect can be  removed.